project-koku/koku
# Better sources processing queue filtering to avoid duplicate events

There was very few checks in the storage layer of the sources client which was causing the Django ORM callback to occur and create Koku-sources sync events even when no values have actually changed.

This change ensures that the process queue has fewer duplicate processing events.

**Testing**
1.  Add OCP Source with many duplicate application and authentication create events.  Verify only a single Source create (POST /provider) occurs.
2. With rabbit down, remove the source and verify sources client recovers from the 500 response and source is removed.  Also verify new sources can be created, showing that the processing queue is healthy.
3. Add a source and remove before it before the provider schema has a chance to complete.

**Test Results**
[sources_process_queue_filter_ut.txt](https://github.com/project-koku/koku/files/3929186/sources_process_queue_filter_ut.txt)
